src/main/java/tel/schich/javacan/BcmMessage.java

```diff
@@ -132,15 +166,15 @@ private BcmMessage(BcmOpcode opcode, @Singular Set<BcmFlag> flags, int count, Bc
                 .putInt(OFFSET_OPCODE, opcode.nativeOpcode)
                 .putInt(OFFSET_FLAGS, BcmFlag.toNative(flags))
                 .putInt(OFFSET_COUNT, count)
-                .putInt(OFFSET_CAN_ID, can_id)
+                .putInt(OFFSET_CAN_ID, canId)
                 .putInt(OFFSET_NFRAMES, frames.size());
-        if (ival1 != null) {
-            putPlatformLong(OFFSET_IVAL1_TV_SEC, ival1.tv_sec);
-            putPlatformLong(OFFSET_IVAL1_TV_USEC, ival1.tv_usec);
+        if (interval1 != null) {
+            putPlatformLong(OFFSET_IVAL1_TV_SEC, interval1.getSeconds());
+            putPlatformLong(OFFSET_IVAL1_TV_USEC, TimeUnit.NANOSECONDS.toMicros(interval1.getNano()));
```

Comment:
It's sad that the Java API completely skipped over micros in pretty much the entire java.time API

Reply:
:man_shrugging: Maybe it is not used that often. I guess millis are used much more often.